components/searchautocomplete.js

```diff
@@ -0,0 +1,40 @@
+import React, { PureComponent } from 'react';
+import Toolbar from 'react-md/lib/Toolbars';
+import Button from 'react-md/lib/Buttons';
+import FontIcon from 'react-md/lib/FontIcons';
+import Drawer from 'react-md/lib/Drawers';
+import AutoComplete from 'react-md/lib/Autocompletes/Autocomplete'
+
+import {connect, dispatch} from 'react-redux'
+import { store, set }  from '../lib/store.js';
+import { search, open } from '../lib/api.js';
+
+function  SearchAutoComplete ({q,  isActive,  results})   {
+  return <AutoComplete
+    id="searchBooks"
+    placeholder="Search"
+    data={results}
+    dataLabel="book"
+    value={q}
+    onChange={ (query) => {
+      search({q: query});
+    }}
+    onAutocomplete = { (workId) => {
+      open({workId});
+    }}
+    block
+    className="md-title--toolbar"
+    inputClassName="md-text-field--toolbar"
+    />;
+}
+
+function mapStateToProps(state) {
+  let props = {
+    isActive: state.getIn('search', 'isActive'),
```

Comment:
`state.getIn('search', 'isActive')` should probably be `state.getIn(['search', 'isActive`])`